### PR TITLE
#169889384-Handle User Verification

### PR DESCRIPTION
**Pull request by ngireric123**
#### What does this PR do?
* Handle User Verification
#### Description of Task to be completed?
* Created component userVerify
* Created test for userVerify component
#### How should this be manually tested?
* Clone ft-user-verification-169889384
* run npm i
* run npm run dev
* navigate to postman
* Register a new user
* after making registration go in your email
* click on verify button

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#169889384
#### Screenshots (if appropriate)
image

image

image

#### Questions:
N/A

### CODE REVIEW

**Review by nignanthomas — CHANGES_REQUESTED** [review]
Well done @ngireric123 
However, as per Miriam requirements, put the backend base Url in a helper named for example `backendCall.js` as we did.

And kindly implement a spinner/loader when to display while the request is pending.

Thanks

File: src/index.html [hunk 1]
```diff
@@ -5,6 +5,8 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <title>Barefoot Nomad</title>
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.9.0/css/all.css" integrity="sha384-i1LQnF23gykqWXg6jxC2ZbCbUMxyw5gLZY6UiUS98LYV5unm8GWmfkIS6jqJfb4E" crossorigin="anonymous">
```
Comment (Cheza-Dzabala) [inline src/index.html]: Swap this with [this](https://github.com/FortAwesome/react-fontawesome) font-awesome react package.

Comment (Cheza-Dzabala) [inline src/index.html]: Swap this with [this](https://github.com/FortAwesome/react-fontawesome) font-awesome react package.

File: src/helpers/backendCall.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import axios from 'axios';
+
+export default axios.create({
+  baseURL: 'https://caret-bn-backend-staging.herokuapp.com/api/v1',
```
Comment (Cheza-Dzabala) [inline src/helpers/backendCall.js]: Export this url to a utility file and import it. That file will hold the endpoints and make it possible to change endpoints based on the environment.

File: src/actions/verifyUserAction.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable import/prefer-default-export */
+import backendCall from '../helpers/backendCall';
+import {
+  VERIFY_SUCCESS, VERIFY_ERROR,
+} from './types';
+
+const verifyType = (type, payload) => ({ type, payload });
+export const VerifyUsers = (token) => (dispatch) => {
```
Comment (Cheza-Dzabala) [inline src/actions/verifyUserAction.js]: Put a new line here.

File: src/actions/verifyUserAction.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+/* eslint-disable import/prefer-default-export */
+import backendCall from '../helpers/backendCall';
+import {
```
Comment (Cheza-Dzabala) [inline src/actions/verifyUserAction.js]: Put a new line here

**Review by Cheza-Dzabala — COMMENTED** [review]
Good job but please see comments made

**Review by Cheza-Dzabala — APPROVED** [review]

**Review by nignanthomas — CHANGES_REQUESTED** [review]
Great job @ngireric123 . However:

- Put your `userVerifyActions` in `authActions.js`.
- Remove `pending` from `reducers`, `actions` and `wherever you are testing it`.

**Review by nignanthomas — APPROVED** [review]

**Pull request merged**